src/nci_util.c

```diff
@@ -912,6 +1040,173 @@ nci_discovery_ntf_copy(
+            case NCI_MODE_PASSIVE_POLL_F:
+            case NCI_MODE_ACTIVE_POLL_F:
+            case NCI_MODE_PASSIVE_POLL_15693:
+            case NCI_MODE_PASSIVE_LISTEN_A:
+            case NCI_MODE_PASSIVE_LISTEN_B:
+            case NCI_MODE_PASSIVE_LISTEN_F:
+            case NCI_MODE_ACTIVE_LISTEN_A:
+            case NCI_MODE_ACTIVE_LISTEN_F:
+            case NCI_MODE_PASSIVE_LISTEN_15693:
+                break;
+            }
+            break;
+        case NCI_RF_INTERFACE_FRAME:
+            /* There are no Activation Parameters for Frame RF interface */
+            break;
+        case NCI_RF_INTERFACE_NFC_DEP:
+            switch (mode) {
+            case NCI_MODE_ACTIVE_POLL_A:
+            case NCI_MODE_ACTIVE_POLL_F:
+            case NCI_MODE_PASSIVE_POLL_A:
+            case NCI_MODE_PASSIVE_POLL_F:
+                nfc_dep_poll = &param->nfc_dep_poll;
+                GASSERT(!nfc_dep_poll->g.bytes || nfc_dep_poll->g.size);
+                GASSERT(!nfc_dep_poll->g.size || nfc_dep_poll->g.bytes);
+                size += G_ALIGN8(nfc_dep_poll->g.size);
+                copy = g_malloc0(size);
+                memcpy(copy, param, sizeof(NciActivationParam));
+                if (nfc_dep_poll->g.bytes) {
+                    guint8* dest = (guint8*)copy +
+                        G_ALIGN8(sizeof(NciActivationParam));
+                    memcpy(dest, nfc_dep_poll->g.bytes,
+                        nfc_dep_poll->g.size);
+                    copy->nfc_dep_poll.g.bytes = dest;
+                }
+                return copy;
+            case NCI_MODE_ACTIVE_LISTEN_A:
+            case NCI_MODE_ACTIVE_LISTEN_F:
+            case NCI_MODE_PASSIVE_LISTEN_A:
+            case NCI_MODE_PASSIVE_LISTEN_F:
+                nfc_dep_listen = &param->nfc_dep_listen;
+                GASSERT(!nfc_dep_listen->g.bytes || nfc_dep_listen->g.size);
+                GASSERT(!nfc_dep_listen->g.size || nfc_dep_listen->g.bytes);
+                size += G_ALIGN8(nfc_dep_listen->g.size);
+                copy = g_malloc0(size);
+                memcpy(copy, param, sizeof(NciActivationParam));
+                if (nfc_dep_listen->g.bytes) {
+                    guint8* dest = (guint8*)copy +
+                        G_ALIGN8(sizeof(NciActivationParam));
+                    memcpy(dest, nfc_dep_listen->g.bytes,
+                        nfc_dep_listen->g.size);
+                    copy->nfc_dep_listen.g.bytes = dest;
+                }
+                return copy;
+            case NCI_MODE_PASSIVE_POLL_B:
+            case NCI_MODE_PASSIVE_POLL_15693:
+            case NCI_MODE_PASSIVE_LISTEN_B:
+            case NCI_MODE_PASSIVE_LISTEN_15693:
+                break;
+            }
+            break;
+        case NCI_RF_INTERFACE_NFCEE_DIRECT:
+        case NCI_RF_INTERFACE_PROPRIETARY:
+            GDEBUG("Unhandled interface type");
+            break;
+        }
+        return NULL;
+    } else {
+        return NULL;
+    }
```

Comment:
Maybe just one `return NULL;` at the end?

Reply:
Sounds reasonable, I have fixed that

Reply:
No, I think you didn't

Reply:
Sorry for that, I have fixed same issue but in other part of the code, now everything should be ok